Dockerfile

```diff
@@ -1,14 +1,21 @@
-FROM ubuntu:latest
-RUN apt update
-# Avoid prompt for locale.
-RUN DEBIAN_FRONTEND="noninteractive" apt install -y tzdata
-RUN apt install -y git netcat wget
-RUN apt install -y openjdk-11-jdk maven
-# Used for waiting for services before starting Olog.
-RUN apt install -y wait-for-it
+# Use Maven image to execute build.
+FROM maven:3.6.3-openjdk-11 AS maven-build
 RUN mkdir olog-es
 WORKDIR /olog-es
 COPY . .
 RUN mvn clean install -DskipTests=true -Pdeployable-jar
+
+# Use smaller openjdk image for running.
+FROM openjdk:11
+# apt clean is run automatically in debian-based images.
+RUN apt update && apt install -y wait-for-it
+# Run commands as user 'olog'
+RUN useradd -ms /bin/bash olog
+# Use previous maven-build image.
+COPY --from=maven-build /olog-es/target /olog-target
+RUN chown olog:olog /olog-target
```

Comment:
Just a minor thing. This `RUN` instruction adds another layer in the image. It's not a big issue, but it's good practice to minimize the amount of layers. Since [COPY](https://docs.docker.com/engine/reference/builder/#copy) supports the `--chown` flag, we could easily include this in the line above:
`COPY --from=maven-build --chown=olog:olog /olog-es/target /olog-target`